ground/src/main/java/com/google/android/ground/model/task/Condition.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.model.task
+
+/** The task ID. */
+typealias TaskId = String
+/** The selected option ID for each task. */
+typealias OptionId = String
+/** The selected options keyed by task ID. */
+typealias TaskSelections = Map<TaskId, Set<OptionId>>
```

Comment:
Nit: Can we name or rescope this in a way that makes its scope and derivation clearer? I understand this to be a "map of the ids of selected options for all preceding multiple choice question tasks, keyed task ID". Since "Condition" isn't conceptually limited to multiple choice questions even though its implementation is, should we instead just pass in the map of all task results, and let the condition figure out which ones it cares about as part of the expression eval logic?

Reply:
Good call, using `Value` instead, to make a direct map from task ID to what we store in the `data` field `DataCollectionViewModel`.